unit_test/degrib2c.c

```diff
@@ -0,0 +1,168 @@
+                numlocal,numfields);
+
+         for (n=0;n<numfields;n++) {
+           ierr=g2_getfld(cgrib,n+1,unpack,expand,&gfld);
+           if (ierr != 0) {
+             printf(" ERROR extracting field = %d\n",ierr);
+             continue;
+           }
+
+           printf("\n  FIELD  %ld\n",n+1);
+           if (n==0) {
+             printf("  SECTION 0:  %ld %ld\n",gfld->discipline,gfld->version);
+             printf("  SECTION 1:  ");
+             for (j=0;j<gfld->idsectlen;j++) printf("%ld ",gfld->idsect[j]);
+             printf("\n");
+           }
+           if ( gfld->local!=0 && gfld->locallen>0 ) {
+              printf("  SECTION 2: %d bytes\n",gfld->locallen);
+              /*for (j=0;j<gfld->locallen;j++) printf("%c",gfld->local[j]);
+              printf("\n");*/
+           }
+           printf("  SECTION 3:  %ld %ld %ld %ld %ld\n",
+                    gfld->griddef,gfld->ngrdpts,gfld->numoct_opt,
+                    gfld->interp_opt,gfld->igdtnum);
+           printf("  GRID TEMPLATE 3. %ld :  ",gfld->igdtnum);
+           for (j=0;j<gfld->igdtlen;j++) printf("%ld ",gfld->igdtmpl[j]);
+           printf("\n");
+           if ( gfld->num_opt == 0 ) 
+             printf("  NO Optional List Defining Number of Data Points.\n");
+           else {
+             printf("  Section 3 Optional List: ");
+             for (j=0;j<gfld->num_opt;j++) printf("%ld ",gfld->list_opt[j]);
+             printf("\n");
+           }
+           printf("  PRODUCT TEMPLATE 4. %ld :  ",gfld->ipdtnum);
+           for (j=0;j<gfld->ipdtlen;j++) printf("%ld ",gfld->ipdtmpl[j]);
+           printf("\n");
+           if ( gfld->num_coord == 0 )
+             printf("  NO Optional Vertical Coordinate List.\n");
+           else {
+             printf("  Section 4 Optional Coordinates:  ");
+             for (j=0;j<gfld->num_coord;j++) printf("%10.6f ",gfld->coord_list[j]);
+             printf("\n");
+           }
+
+           if ( gfld->ibmap != 255 )
+              printf("  Num. of Data Points =  %ld     with BIT-MAP %ld\n",
+                     gfld->ndpts,gfld->ibmap);
+           else
+              printf("  Num. of Data Points =  %ld     NO BIT-MAP\n",gfld->ndpts);
+
+           printf("  DRS TEMPLATE 5. %ld :  ",gfld->idrtnum);
+           for (j=0;j<gfld->idrtlen;j++) printf("%ld ",gfld->idrtmpl[j]);
+           printf("\n");
+           //printf(" SAGlist:   ");
+           //for (j=0;j<100;j++) printf("%f ",gfld->fld[j]);
+           //printf("\n");
+
+           fldmax=gfld->fld[0];
+           fldmin=gfld->fld[0];
+           sum=gfld->fld[0];
+           for (j=1;j<gfld->ndpts;j++) {
+             if (gfld->fld[j] > fldmax) fldmax=gfld->fld[j];
+             if (gfld->fld[j] < fldmin) fldmin=gfld->fld[j];
+             sum=sum+gfld->fld[j];
+           }
+           printf("  Data Values:\n");
+           printf("  MIN=%21.8f  AVE=%21.8f  MAX=%21.8f\n",
+                  fldmin,sum/gfld->ndpts,fldmax);
```

Comment:
If there are many data values, you can't encode them all as constants in the test file, but you can check a few actual values.